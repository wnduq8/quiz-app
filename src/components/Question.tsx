import Radio from '@components/atomic/Radio'
import styled from 'styled-components'

interface IQuestionProps {
  question: string
  questionList: string[]
  userAnswer: string
  onChange?: (value: string) => void
}

function Question({ question, questionList, userAnswer, onChange }: IQuestionProps) {
  return (
    <QuestionBlock>
      <p>{question}</p>
      <ul>
        {questionList.map((question: string, index: number) => (
          <li key={question}>
            <Radio
              value={question}
              onChange={(e) => onChange && onChange(e.target.value)}
              checked={userAnswer === question}
              label={question}
              id={`${question}-${index + 1}`}
            />
          </li>
        ))}
      </ul>
    </QuestionBlock>
  )
}

export default Question

const QuestionBlock = styled.section`
  > ul {
    display: flex;
    flex-direction: column;
    row-gap: 10px;

    li {
      label {
        padding-left: 10px;
      }

      input,
      label {
        cursor: pointer;
      }
    }
  }
`
