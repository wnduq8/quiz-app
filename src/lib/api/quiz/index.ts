import { CommonResult, IGetQuizParams, IQuizRes } from '@lib/api/types'
import httpClient from '@lib/api/httpClient'

export async function getQuiz(params: IGetQuizParams) {
  const response = await httpClient.get<CommonResult<IQuizRes[]>>('/api.php', { params })
  return response.data
}
