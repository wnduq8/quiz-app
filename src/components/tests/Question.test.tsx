import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Question from '@components/Question'
import { fakeQuiz } from '@src/tests/quiz'

const { question, question_list, correct_answer } = fakeQuiz

describe('Question', () => {
  it('Snapshot - 항목을 선택했을 때', () => {
    const questionComponent = renderer.create(
      <Question question={question} questionList={question_list} userAnswer={correct_answer} />,
    )

    expect(questionComponent.toJSON()).toMatchSnapshot()
  })

  it('Snapshot - 항목을 선택하지 않았을 때', () => {
    const questionComponent = renderer.create(
      <Question question={question} questionList={question_list} userAnswer={''} />,
    )

    expect(questionComponent.toJSON()).toMatchSnapshot()
  })
})
