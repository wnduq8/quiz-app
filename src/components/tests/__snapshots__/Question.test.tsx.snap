// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Snapshot - 항목을 선택하지 않았을 때 1`] = `
<section
  className="sc-bcXHqe LzPpX"
>
  <p>
    Which game did NOT get financed via Crowdfunding?
  </p>
  <ul>
    <li>
      <input
        checked={false}
        id="Town of Salem-1"
        onChange={[Function]}
        type="radio"
        value="Town of Salem"
      />
      <label
        htmlFor="Town of Salem-1"
      >
        Town of Salem
      </label>
    </li>
    <li>
      <input
        checked={false}
        id="Undertale-2"
        onChange={[Function]}
        type="radio"
        value="Undertale"
      />
      <label
        htmlFor="Undertale-2"
      >
        Undertale
      </label>
    </li>
    <li>
      <input
        checked={false}
        id="Tower Unite-3"
        onChange={[Function]}
        type="radio"
        value="Tower Unite"
      />
      <label
        htmlFor="Tower Unite-3"
      >
        Tower Unite
      </label>
    </li>
    <li>
      <input
        checked={false}
        id="Enter the Gungeon-4"
        onChange={[Function]}
        type="radio"
        value="Enter the Gungeon"
      />
      <label
        htmlFor="Enter the Gungeon-4"
      >
        Enter the Gungeon
      </label>
    </li>
  </ul>
</section>
`;

exports[`Question Snapshot - 항목을 선택했을 때 1`] = `
<section
  className="sc-bcXHqe LzPpX"
>
  <p>
    Which game did NOT get financed via Crowdfunding?
  </p>
  <ul>
    <li>
      <input
        checked={false}
        id="Town of Salem-1"
        onChange={[Function]}
        type="radio"
        value="Town of Salem"
      />
      <label
        htmlFor="Town of Salem-1"
      >
        Town of Salem
      </label>
    </li>
    <li>
      <input
        checked={false}
        id="Undertale-2"
        onChange={[Function]}
        type="radio"
        value="Undertale"
      />
      <label
        htmlFor="Undertale-2"
      >
        Undertale
      </label>
    </li>
    <li>
      <input
        checked={false}
        id="Tower Unite-3"
        onChange={[Function]}
        type="radio"
        value="Tower Unite"
      />
      <label
        htmlFor="Tower Unite-3"
      >
        Tower Unite
      </label>
    </li>
    <li>
      <input
        checked={true}
        id="Enter the Gungeon-4"
        onChange={[Function]}
        type="radio"
        value="Enter the Gungeon"
      />
      <label
        htmlFor="Enter the Gungeon-4"
      >
        Enter the Gungeon
      </label>
    </li>
  </ul>
</section>
`;
